### ISSUE TITLE
Malware Patrol FREE can also be a product code of 32 (maybe others?)

### ISSUE THREAD

**Issue by Jas0n99** [extremeshok/clamav-unofficial-sigs#298]
This has been a long-standing bug (which has been mentioned before but closed without resolve)...

Starting around 2220 all the malware patrol code focuses on the "free" version assuming a product code of 8, and that is FALSE. For years now a free product code could also be 32 (maybe others?)... If you set your user.conf as follows:

```
malwarepatrol_free="yes"
malwarepatrol_list="clamav_basic"
malwarepatrol_product_code="32"
malwarepatrol_receipt_code="xxxxx"
```
The script thinks it's non-free, forces the product code to 8. The file that is downloaded and written to /var/lib/clamav/malwarepatrol.db only has the following in it:

```
# Your access was denied.
# You may have supplied a wrong password, your subscription may have expired or you may not have access to this resource.

```
It doesn't throw an error downloading because there are only comments in the file. :(

Honestly there is no reason to be forcing `malwarepatrol_free`, `malwarepatrol_list`, or `malwarepatrol_product_code` as it currently is making the wrong assumptions and breaking this feature for people (some may not even realize it!). It would be better to do some additional checking after downloading the file to ensure it's not just the above comments (rule checking, file size, whatever).

**Comment by razorsedge** [comment]
This is due to #291 .  I have the exact same settings as you.

I do agree that the docs and/or code need to be updated to deal with this.

**Comment by Jas0n99** [comment]
Yes I noticed that bug too and fixed it while cleaning up this issue.

**Comment by perplexityjeff** [comment]
We had this issue as well and the fix suggested in #291 did indeed fix the issue.

**Comment by extremeshok** [comment]
fixed in dev
